lib/ash/type/union.ex

```diff
@@ -521,8 +521,12 @@ defmodule Ash.Type.Union do
          {:ok, type_config} <- Keyword.fetch(type_configs, type_name),
          {:ok, type} <- Keyword.fetch(type_config, :type),
          type_constraints <- Keyword.get(type_config, :constraints, []),
-         {:ok, new_value} <- type.handle_change(nil, new_value, type_constraints) do
-      {:ok, %Ash.Union{type: type_name, value: new_value}}
+         true <- :erlang.function_exported(type, :handle_change, 3),
```

Comment:
I think what we need here instead is `type <- Ash.Type.get_type(type)`, which will convert `:string` to `Ash.Type.String`.

Reply:
@zachdaniel resolved